src/main/java/uk/co/real_logic/benchmarks/rtt/aeron/LiveRecordingMessageTransceiver.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    LiveRecordingMessageTransceiver(
+        final ArchivingMediaDriver archivingMediaDriver,
+        final AeronArchive aeronArchive,
+        final boolean ownsArchiveClient,
+        final MessageRecorder messageRecorder)
+    {
+        super(messageRecorder);
+        this.archivingMediaDriver = archivingMediaDriver;
+        this.aeronArchive = aeronArchive;
+        this.ownsArchiveClient = ownsArchiveClient;
+    }
+
+    public void init(final Configuration configuration) throws Exception
+    {
+        final AeronArchive.Context context = aeronArchive.context();
+        final Aeron aeron = context.aeron();
+
+        frameCountLimit = frameCountLimit();
+
+        subscription = aeron.addSubscription(receiveChannel(), receiveStreamId());
+
+        final String sendChannel = sendChannel();
+        final int sendStreamId = sendStreamId();
+        publication = aeron.addExclusivePublication(sendChannel, sendStreamId);
+
+        final int publicationSessionId = publication.sessionId();
+        final String channel = addSessionId(sendChannel, publicationSessionId);
+        aeronArchive.startRecording(channel, sendStreamId, LOCAL, true);
+
+        recordingEventsSubscription = aeron.addSubscription(context.recordingEventsChannel(), context
+            .recordingEventsStreamId());
+
+        recordingEventsAdapter = new RecordingEventsAdapter(new RecordingEventsListener()
+        {
+            public void onStart(
+                final long recordingId,
+                final long startPosition,
+                final int sessionId,
+                final int streamId,
+                final String channel,
+                final String sourceIdentity)
+            {
+            }
+
+            public void onProgress(final long recordingId, final long startPosition, final long position)
+            {
+                if (recordingId == LiveRecordingMessageTransceiver.this.recordingId)
+                {
+                    if (NULL_POSITION == recordingPositionConsumed)
+                    {
+                        recordingPositionConsumed = startPosition;
+                    }
+                    LiveRecordingMessageTransceiver.this.recordingPosition = position;
+                }
+            }
+
+            public void onStop(final long recordingId, final long startPosition, final long stopPosition)
+            {
+                if (recordingId == LiveRecordingMessageTransceiver.this.recordingId)
+                {
+                    LiveRecordingMessageTransceiver.this.recordingPosition = stopPosition;
+                }
+            }
+        }, recordingEventsSubscription, frameCountLimit);
+
+        recordingId = awaitRecordingStart(aeron, publicationSessionId);
+
+        while (!recordingEventsSubscription.isConnected() || !subscription.isConnected() || !publication.isConnected())
+        {
+            Thread.yield();
```

Comment:
Check for interrupt.

Reply:
Now using utility method which throws `IllegalStateException` if thread is interrupted.